### PR TITLE
Adding datepicker API doc

### PR DESCRIPTION
**Pull request by dusave**
Describe your changes here.

Adds the Datepicker API doc

### Screenshots

Base

### Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

**Comment by siddharthkp** [comment]
Played around with the [storybook from the other PR](https://primer-components-1015mjqb0-primer.vercel.app/react/storybook?path=/story/composite-components-datepicker--start-day-picker), this is so well built!

Initial thoughts on API:

There is a wip ADR that talks about the API direction https://github.com/primer/react/pull/1703, you can see some of this in action in Autocomplete and [ActionMenu v2](https://primer.style/react/drafts/ActionMenu2)

When there are props on the top level component `DatePicker` which are passed down to children like `anchoredOverlayProps` and `iconPlacement`, we should expose that detail in the API as compound components: `DatePicker.Button` and `DatePicker.Overlay`

```jsx
<DatePicker>
  <DatePicker.Button>Choose Date</DatePicker.Button>
  <DatePicker.Overlay />
<DatePicker>
```

`DatePicker.Overlay`: `anchoredOverlayProps` like `align` and `side` become props for `DatePicker.Overlay`.
`DatePicker.Button`: `placeholder` becomes `children`, `iconPlacement` gets replaced with `leadingIcon` and `trailingIcon` based on [new button props](https://primer-components-65wwy51f1-primer.vercel.app/react/NewButton#icon-and-button). For this component, `leadingIcon` has a default value of `CalendarIcon`

What do you think?

**Comment by dusave** [comment]
@siddharthkp Thanks for taking a look!

I really like the idea of using the compound props for this, it would definitely simplify the more complex implementations. One concern I do have though is with the simpler implementation, where the consumer just wants the basic datepicker, no props. Would that require them to insert the `Button` and `Overlay` components as children, even when they don't want to override any of the default behavior?

File: docs/content/Datepicker.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+title: Datepicker
+description: Use datepicker to select a single date, multiple dates, or a range of dates
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/Datepicker/index.tsx
+storybook: '/react/storybook?path=/story/composite-components-datepicker--default-date-picker'
+---
+
+## Default example
```
Comment (mperrotti) [inline docs/content/Datepicker.md:9]: Minor nitpick: let's structure the headings to match what has been defined in https://github.com/primer/react/issues/1701

**Comment by rezrah** [comment]
Hi @dusave - sorry it's taken so long to get back to you on this. I've spoken to a few Primer folks about our roadmap and unfortunately a date picker isn't on the near-term agenda for patterns that we plan to support. Reason being that the level of niche product focus and complexity balanced against the maintenance overhead means we'll struggle to support it. 

Do you see any issues in keeping this component internal for the time-being @dusave?. This is just the current state of play too, so we can revisit this decision down the line. 

cc. @gaknoll

**Pull request closed without merging**